packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts

```diff
@@ -275,6 +284,27 @@ export class AmplifyGraphqlApi extends Construct {
     outputStorageStrategy.addBackendOutputEntry(graphqlOutputKey, output);
   }
 
+  /**
+   * Gets the deployment information from the context.
+   * @returns the deployment identifier.
+   */
+  private getDeploymentIdentifier(): DeploymentIdentifier {
+    const deploymentType = this.node.tryGetContext(CDKContextKey.DEPLOYMENT_TYPE);
+
+    if (deploymentType === 'sandbox') {
+      return {
+        deploymentType: deploymentType,
+        namespace: this.node.tryGetContext(CDKContextKey.BACKEND_NAMESPACE),
+      } as SandboxDeploymentIdentifier;
+    } else {
+      return {
+        deploymentType: deploymentType,
+        namespace: this.node.tryGetContext(CDKContextKey.BACKEND_NAMESPACE),
+        name: this.node.tryGetContext(CDKContextKey.BACKEND_NAME),
```

Comment:
What if one or both of the context key isn't set? Maybe it's always set for a Gen 2 CLI invocation, but customers can also do a straight-up `cdk synth` or `cdk deploy` without setting context.